src/test/java/model/TestCodeQualityRuleModel.java

```diff
@@ -0,0 +1,124 @@
+package test.java.model;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotEquals;
+
+import org.junit.jupiter.api.Test;
+
+import main.java.model.CodeQualityRule;
+
+/**
+ * @author Lino Silva
+ *
+ */
+class TestCodeQualityRuleModel {
+
+	/**
+	 * This test validates if a default CodeQualityRule is being created successfully
+	 */
+	@Test
+	void testCreationOfDefaultRule() {
+		CodeQualityRule ruleDefault = new CodeQualityRule("is_long_method", "LOC > 80 && CYCLO > 10", true, false);
+		
+		assertEquals("is_long_method", ruleDefault.getName());
+		assertEquals("LOC > 80 && CYCLO > 10", ruleDefault.getRule());
+		assertEquals(true, ruleDefault.isDefault());
+		assertEquals(false, ruleDefault.isAdvanced());
+	}
+	
+	/**
+	 * This test validates if a non default CodeQualityRule is being created successfully
+	 */
+	@Test
+	void testCreationOfNonDefaultRule() {
+		CodeQualityRule ruleNonDefault = new CodeQualityRule("is_banana", "ATFD > 4 && LAA < 0.42", false, false);
+		
+		assertEquals("is_banana", ruleNonDefault.getName());
+		assertEquals("ATFD > 4 && LAA < 0.42", ruleNonDefault.getRule());
+		assertEquals(false, ruleNonDefault.isDefault());
+		assertEquals(false, ruleNonDefault.isAdvanced());
+	}
+	
+	/**
+	 * This test verifies if the name of a default rule can be changed with the 
+	 * setName method
+	 */
+	@Test
+	void testNameChangeDefaultRule() {
+		CodeQualityRule ruleDefault = new CodeQualityRule("is_long_method", "LOC > 80 && CYCLO > 10", true, false);
+		
+		assertEquals("is_long_method", ruleDefault.getName());
+		ruleDefault.setName("new_method");
+		assertNotEquals("new_method", ruleDefault.getName());
+		assertEquals("is_long_method", ruleDefault.getName());
+	}
+	
+	/**
+	 * This test verifies if the rule content of a default rule can be changed with the 
+	 * setRule method
+	 */
+	@Test
+	void testRuleChangeDefaultRule() {

```

Comment:
I get what you were going for mas acho que testRuleEditionDefaultRule() ficava mais fácil de perceber tbh